### PR TITLE
Add support for the `ALTER USER` statement to change account passwords

### PR DESCRIPTION
**Pull request by fulghum**
Adds basic support for the `ALTER USER` statement, so that users can change passwords. All users are allowed to change their own password; changing another user's password requires the `CREATE USER` privilege or the `UPDATE` privilege on the `mysql` database.

[MySQL reference docs for `ALTER USER`](https://dev.mysql.com/doc/refman/8.0/en/alter-user.html)

Related to: https://github.com/dolthub/dolt/issues/7348

### CODE REVIEW

File: sql/planbuilder/priv.go [hunk 1]
```diff
@@ -147,30 +147,39 @@ func convertPrivilegeLevel(privLevel ast.PrivilegeLevel) plan.PrivilegeLevel {
...
+		if user.Auth1.Plugin == "mysql_native_password" && len(user.Auth1.Password) > 0 {
+			authUser.Auth1 = plan.AuthenticationMysqlNativePassword(user.Auth1.Password)
+		} else if len(user.Auth1.Plugin) > 0 {
+			authUser.Auth1 = plan.NewOtherAuthentication(user.Auth1.Password, user.Auth1.Plugin)
+		} else {
+			// We default to using the password, even if it's empty
+			authUser.Auth1 = plan.NewDefaultAuthentication(user.Auth1.Password)
+		}
+	}
+	if user.Auth2 != nil || user.Auth3 != nil || user.AuthInitial != nil {
```
Comment (max-hoffman) [inline sql/planbuilder/priv.go:165]: if know you didn't write this and I don't know if this scenario is possible, but if there are multiple auth options this fails even if Auth1 is OK

**Review by max-hoffman — APPROVED** [review]
lgtm, I'm not too familiar with the auth execution code but seems reasonable to me

File: sql/planbuilder/priv.go (hunk 1 above)
Comment (fulghum) [reply sql/planbuilder/priv.go:166]: I dug into this a little bit and it looks like this is by design. This line is the only reference to `Auth2`, `Auth3`, and `AuthInitial` in GMS, so this seems to just be ensuring that only `Auth1` is ever used. 

I went ahead and added a comment mentioning that, too.

File: sql/plan/alter_user.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+var _ sql.CollationCoercible = (*AlterUser)(nil)
+
+// Schema implements the interface sql.Node.
+func (a *AlterUser) Schema() sql.Schema {
+	return types.OkResultSchema
+}
+
+// String implements the interface sql.Node.
+func (a *AlterUser) String() string {
+	ifNotExists := ""
```
[the remark this answers is not in the document]
Comment (fulghum) [reply sql/plan/alter_user.go]: Ah! good catch! That was copied from `CREATE USER` and I missed flipping that var name. Fixed in 7b7c329cdea598b9b98767fc0c9d5e9edcb81639.

File: enginetest/queries/priv_auth_queries.go [hunk 3]
```diff
@@ -2147,6 +2147,160 @@ func (p *NoopPlaintextPlugin) Authenticate(db *mysql_db.MySQLDb, user string, us
...
+				Query:       "CREATE USER `user2`@`localhost` IDENTIFIED BY '';",
+				ExpectedErr: false,
+			}, {
+				Username:    "root",
+				Password:    "",
+				Query:       "GRANT SELECT ON *.* TO `user2`@`localhost`;",
+				ExpectedErr: false,
+			},
+
+			// When IF EXISTS is specified, an error isn't returned if the user doesn't exist
```
[the remark this answers is not in the document]
Comment (fulghum) [reply enginetest/queries/priv_auth_queries.go:2205]: Good idea. Added in 7b7c329cdea598b9b98767fc0c9d5e9edcb81639.

**Pull request merged**